### PR TITLE
Add Carthage and CocoaPods support for microframeworks

### PR DESCRIPTION
**Pull request by knothed**
When importing JamitFoundation via SPM, the user can choose which microframeworks (PageView, CarouselView etc.) should be loaded. Then, in code, the user can `import JamitFoundation` to load JamitFoundation's core functionality, and `import PageView` to import the microframework PageView.

This PR mirrors this functionality with Carthage. Now, when importing JamitFoundation via Carthage, all microframeworks are loaded, and the user can import them by `import PageView`, just like when using SPM.

There is no such functionality with CocoaPods. Instead of defining in code which microframeworks to use (using `import`s), this is defined in the Podfile:
- `pod JamitFoundation/Core` will only load JamitFoundation's core
- `pod JamitFoundation/PageView` will load JamitFoundation's core and the microframework PageView
- `pod JamitFoundation` will load JamitFoundation with *all* of its microframeworks.

Then, in code, you just `import JamitFoundation` to import all functionality that has been loaded.

---
Corresponds to https://git.jamitlabs.net/jamit-labs/trockner/-/issues/519.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,19 @@ Add the following line to your Cartfile.
 git "git@github.com:JamitLabs/JamitFoundation.git"
 ```
 
+### CocoaPods
+
+Add the following line to your Podfile.
+
+```swift
+pod 'JamitFoundation', :git => 'https://github.com/knothed/JamitFoundation.git', :tag => '1.4.2', :inhibit_warnings => true
```
Comment (SimonNumberTwo) [inline README.md:40]: Why is there `knothed` mentioned here? Is there something missing in the main repository so you needed to fork it?

File: README.md [hunk 2]
```diff
@@ -32,6 +32,19 @@ Add the following line to your Cartfile.
...
+```
+
+If you don't want to include all of JamitFoundation, but only specific microframeworks (e.g. PageView), add the following line to your Podfile.
+
+```swift
+pod 'JamitFoundation/PageView', :git => 'https://github.com/knothed/JamitFoundation.git', :tag => '1.4.2', :inhibit_warnings => true
```
Comment (SimonNumberTwo) [inline README.md:46]: Same as above, check the link as it contains `knothed`

File: Examples/JamitFoundationExample/JamitFoundationExample.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -834,7 +839,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				DEVELOPMENT_TEAM = DB7EUU9D79;
 				INFOPLIST_FILE = App/SupportingFiles/Info.plist;
-				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Comment (SimonNumberTwo) [inline Examples/JamitFoundationExample/JamitFoundationExample.xcodeproj/project.pbxproj:842]: Are we sure we want to increase the deployment target? Wasn't it even 9.0 before?

File: Examples/JamitFoundationExample/JamitFoundationExample.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -853,7 +858,7 @@
...
```
Comment (SimonNumberTwo) [inline Examples/JamitFoundationExample/JamitFoundationExample.xcodeproj/project.pbxproj:861]: See comment above for the deployment target. We need 9.0 for a specific app we are supporting.

**Review by SimonNumberTwo — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (knothed) [reply README.md:40]: I forgot to change it :D after the MR, of course the main repository should be referenced.

File: Examples/JamitFoundationExample/JamitFoundationExample.xcodeproj/project.pbxproj (hunk 4 above)
Comment (knothed) [reply Examples/JamitFoundationExample/JamitFoundationExample.xcodeproj/project.pbxproj:861]: Okay, but this deployment target is actually **only** for the example project. Therefore it should be irrelevant for any app which uses JamitFoundation (the JamitFoundation deployment target didn't change).

File: Examples/JamitFoundationExample/JamitFoundationExample.xcodeproj/project.pbxproj (hunk 3 above)
Comment (knothed) [reply Examples/JamitFoundationExample/JamitFoundationExample.xcodeproj/project.pbxproj:842]: This deployment target also only applies to the example project, not to JamitFoundation itself.

**Review by SimonNumberTwo — APPROVED** [review]

**Comment by knothed** [comment]
@SimonNumberTwo could you merge? I don’t have merge access.

**Pull request merged**